client/lib/wp/sync-handler/whitelist-handler.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Module dependencies
+ */
+
+import debugFactory from 'debug';
+const debug = debugFactory( 'calypso:sync-handler:whitelist' );
+
+const whitelist = [
+	'^/me/posts$',
+	'^/me/settings',
+	'^/sites/[\\w.]+/posts'
+];
+
+export const isWhitelisted = params => {
+	const { path } = params;
+	let whitelisted = false;
+
+	if ( params.method && 'get' !== params.method.toLowerCase() ) {
+		debug( 'No allow %o request', params.method );
+		return false;
+	};
+
+	for ( let i = 0; i < whitelist.length; i++ ) {
+		let item = whitelist[ i ];
+		let regExp = new RegExp( item );
+		if ( regExp.test( path ) ) {
+			debug( '%o whitelisted', path );
+			whitelisted = true;
```

Comment:
can we break the bucle `for` with a return or will keep walking through of the whole array.